### PR TITLE
bpo-43321: Fix SystemError in getargs.c

### PR DESCRIPTION
**Pull request by methane**
https://bugs.python.org/issue43321

### CODE REVIEW

File: Python/getargs.c [hunk 1]
```diff
@@ -658,9 +658,9 @@ convertsimple(PyObject *arg, const char **p_format, va_list *p_va, int flags,
     /* For # codes */
 #define REQUIRE_PY_SSIZE_T_CLEAN \
     if (!(flags & FLAG_SIZE_T)) { \
-        PyErr_SetString(PyExc_SystemError, \
-                        "PY_SSIZE_T_CLEAN macro must be defined for '#' formats"); \
-        return NULL; \
+        const char *msg = "PY_SSIZE_T_CLEAN macro must be defined for '#' formats"; \
+        PyErr_SetString(PyExc_SystemError, msg); \
+        return msg; \
```
Comment (serhiy-storchaka) [inline Python/getargs.c]: Maybe use `RETURN_ERR_OCCURRED` for consistency with other code? The content of the returned buffer is ignored if an error is set.

**Review by vstinner — APPROVED** [review]
LGTM.

File: Python/getargs.c [hunk 2]
```diff
@@ -660,7 +660,7 @@ convertsimple(PyObject *arg, const char **p_format, va_list *p_va, int flags,
     if (!(flags & FLAG_SIZE_T)) { \
         PyErr_SetString(PyExc_SystemError, \
                         "PY_SSIZE_T_CLEAN macro must be defined for '#' formats"); \
-        return NULL; \
+        RETURN_ERR_OCCURRED; \
     }
 #define RETURN_ERR_OCCURRED return msgbuf
```
Comment (vstinner) [inline Python/getargs.c]: Oh. You may define RETURN_ERR_OCCURRED before using it (move the macro).

**Comment by serhiy-storchaka** [comment]
:+1:

**Pull request merged**